src/cswinrt/strings/additions/Windows.Foundation/Windows.Foundation.cs

```diff
@@ -25,8 +25,10 @@ public static Task AsTask(this IAsyncAction source, CancellationToken cancellati
                 throw new ArgumentNullException(nameof(source));
             }
 
-            // TODO: Handle the scenario where the 'IAsyncAction' is actually a task (i.e. originated from native code
-            // but projected into an IAsyncAction)
+            if (source is ITaskAwareAsyncInfo asyncInfo && asyncInfo.Task is Task task)

```

Comment:
As discussed earlier, we should figure out what this means for the cancellation token.

Reply:
Handled.